File: Directory Space/dirSpace.py
# About
# Simple Python program to calculate the size of a directory and all subdirectories using recursion

import os

def dirSpace(dir):
    # variables
    tempSum = 0
    entries = os.listdir(dir)

    # for each entry
    for entry in entries:

        # get path for the entry
        entryPath = os.path.join(dir, entry)

        # if entry is file, add to size
        if os.path.isfile(entryPath):
            tempSum += os.path.getsize(entryPath)

        # if entry is directory, recurse
        elif os.path.isdir(entryPath):
            tempSum += dirSpace(entryPath)

    return tempSum


def main():
    topDir = os.getcwd()
    size = dirSpace(topDir)
    print(size)


if __name__ == "__main__":
    main()

File: Directory Space/dirSpace.java
//About
//Simple java program to calculate the size of a directory and all subdirectories using recursion

// Importing File Class
import java.io.File;

// class
public class dirSpace {

    // recursive function 
    public static long getSize(File dir) {
        // variable
        long tempSum = 0;

        // if is a file, add to sum
        if (dir.isFile()) {
            tempSum += dir.length();

        // if is a directory, recurse
        } else if (dir.isDirectory()) {
            
            File[] files = dir.listFiles();
            if (files != null) {
                for (File file : files) {
                    tempSum += getSize(file);
                }
            }
        }

        return tempSum;
    }

    public static void main(String[] args) {

        File file = new File(".");
        long size = getSize(file);
        System.out.println(size);

    }
}